packages/flame/lib/src/widgets/animation_widget.dart

```diff
@@ -67,24 +67,66 @@ class SpriteAnimationWidget extends StatelessWidget {
   })  : _animationFuture = SpriteAnimation.load(path, data, images: images),
         _animationTicker = null;
 
+  @override
+  State<SpriteAnimationWidget> createState() => _SpriteAnimationWidgetState();
+}
+
+class _SpriteAnimationWidgetState extends State<SpriteAnimationWidget> {
+  late FutureOr<SpriteAnimation> _animationFuture = widget._animationFuture;
+
+  @override
+  void didUpdateWidget(covariant SpriteAnimationWidget oldWidget) {
+    super.didUpdateWidget(oldWidget);
+
+    _updateAnimation(oldWidget._animationFuture, widget._animationFuture);
+  }
+
+  Future<void> _updateAnimation(
+    FutureOr<SpriteAnimation> oldFutureValue,
+    FutureOr<SpriteAnimation> newFutureValue,
+  ) async {
+    final oldValue = await oldFutureValue;
+    final newValue = await newFutureValue;
+
+    final areFramesDifferent =
+        oldValue.frames.length != newValue.frames.length ||
+            oldValue.frames.fold(
+              false,
+              (previous, frame) {
+                final newFrame =
+                    newValue.frames[oldValue.frames.indexOf(frame)];
+
+                return frame.sprite.image != newFrame.sprite.image ||
+                    frame.sprite.src != newFrame.sprite.src;
```

Comment:
You need to take previous into consideration here right? And then have the default of the fold `true` instead of `false`.

Reply:
Duh 🤦  ofc I need! Thanks!